src/spanningtrees/kruskal.jl

```diff
@@ -6,27 +6,40 @@ end
 isless(e1::KruskalHeapEntry, e2::KruskalHeapEntry) = e1.dist < e2.dist
 
 """
-    quick_find!(vs, p, q)
+    find_set!(vs, q)
 
 Perform [Quick-Find algorithm](https://en.wikipedia.org/wiki/Disjoint-set_data_structure)
-on a given pair of vertices `p`and `q`, and make a connection between them in the vector `vs`.
+on a given vertice `q`, finds the component id it belongs to.
 """
-function quick_find!(vs, p, q)
-    pid = vs[p]
-    qid = vs[q]
-    for i in 1:length(vs)
-        if vs[i] == pid
-            vs[i] = qid
-        end
+
+function find_set!(set_id::Vector{U}, q::U) where U<:Integer
+    
+    root = q
+    while root != set_id[root]
+        root = set_id[root]
+    end
```

Comment:
Why is this a `while` statement as opposed to an `if`?

Reply:
Each set (connected component) is represented by a tree in a forest. The root of each tree represents the set. The while loop is necessary to is to traverse up the tree of the selected node q
and find the root representing the set q belongs to.

Additional [reference](https://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-046j-design-and-analysis-of-algorithms-spring-2015/recitation-notes/MIT6_046JS15_Recitation3.pdf) for the algorithm used

Reply:
I’m confused. The while loop checks for a condition, and then causes the condition to break the loop after the first iteration. root becomes set_id[root] in the first loop, so you will break the while immediately after. This is equivalent to an if.

Reply:
Ah, got it. Thanks.